deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/ConvolutionLayer.java

```diff
@@ -89,14 +89,14 @@ protected ConvolutionLayer(BaseConvBuilder<?> builder) {
         this.hasBias = builder.hasBias;
         this.convolutionMode = builder.convolutionMode;
         this.dilation = builder.dilation;
-        if (builder.kernelSize.length != 2)
-            throw new IllegalArgumentException("Kernel size of should be rows x columns (a 2d array)");
+//        if (builder.kernelSize.length != 2)
+//            throw new IllegalArgumentException("Kernel size of should be rows x columns (a 2d array)");
```

Comment:
Keep the checks - but maybe add a numDimensions() on the builder or something... so builder.kernelSize.length != numDimensions() or whatever.